src/auth/connection.ts

```diff
@@ -23,10 +23,37 @@ export const ssoAccountAccessScopes = ['sso:account:access']
 export const codewhispererScopes = ['codewhisperer:completions', 'codewhisperer:analysis']
 export const defaultSsoRegion = 'us-east-1'
 
+type SsoType =
+    | 'any' // any type of sso
+    | 'base' // AWS Identity Center
+    | 'builderId'
+
 export const isIamConnection = (conn?: Connection): conn is IamConnection => conn?.type === 'iam'
-export const isSsoConnection = (conn?: Connection): conn is SsoConnection => conn?.type === 'sso'
-export const isBuilderIdConnection = (conn?: Connection): conn is SsoConnection =>
-    isSsoConnection(conn) && conn.startUrl === builderIdStartUrl
+export const isSsoConnection = (conn?: Connection, type: SsoType = 'any'): conn is SsoConnection => {
+    if (conn?.type !== 'sso') {
+        return false
+    }
+    // At this point the conn is an SSO conn, but now we must determine the specific type
+    switch (type) {
+        case 'base':
+            // Any SSO type is considered an SSO connection, but in this case
+            // we only want an Identity Center connection. So we return true
+            // as long as the connection is none of the other connection types.
+
+            // IMPORTANT: This condition should grow as more SsoType's get added.
+            return !isBuilderIdConnection(conn)
+        case 'builderId':
+            return conn.startUrl === builderIdStartUrl
+        case 'any':
+            return true
+    }
+}
+export const isAnySsoConnection = (conn?: Connection): conn is SsoConnection => isSsoConnection(conn, 'any')
```

Comment:
`type` defaults to 'any' so is this function needed?

Reply:
This is used in a few places with `filter()` and lets us do `filter(isAnySsoConnection)`, instead of `filter((e) => isSsoConnection(e))`

Reply:
can't `filter(isSsoConnection)` work if `type` defaults to 'any' ?

Reply:
I wish that was the case :(
The argument for `filter()` is `predicate: (value: never, index: number, array: never[])` so the `index` argument will incorrectly be passed to `type`